everestrs/everestrs_sys/everestrs_sys.cpp

```diff
@@ -30,6 +33,41 @@ JsonBlob json2blob(const json& j) {
     return JsonBlob{vec};
 }
 
+// Below are overloads to be used with std::visit and our std::variant. We force
+// a compilation error if someone changes the underlying std::variant without
+// extending/adjusting the functions below.
+
+template <typename T, typename... VARIANT_T> struct VariantMemberImpl : public std::false_type {};
+
+template <typename T, typename... VARIANT_T>
+struct VariantMemberImpl<T, std::variant<VARIANT_T...>> : public std::disjunction<std::is_same<T, VARIANT_T>...> {};
+
+/// @brief Static checker if the type T can be converted to `ConfigEntry`.
+///
+/// We use this to detect `get_config_field` overloads which receive arguments
+/// which aren't part of our `ConfigEntry` variant.
+template <typename T> struct ConfigEntryMember : public VariantMemberImpl<T, ConfigEntry> {};
+
+inline ConfigField get_config_field(const std::string& _name, bool _value) {
```

Comment:
i would probably have created a single class with visitor functions, which gets the name as its constructor argument and whose instance could then be used directly within `std::visit`